src/ManagerV2/UserManager.php

```diff
@@ -136,4 +142,26 @@ private function getNextHomonymUsername(string $baseUsername, array $homonyms):
     {
         return sprintf('%s-%d', $baseUsername, $this->getMaxHomonymIndex($baseUsername, $homonyms) + 1);
     }
+
+    public function checkMfaCode(User $user, ?string $authCode): void
+    {
+        $isMfaEnabled = $user->isMfaEnabled() && $this->appli2faEnabled;
+        if ($isMfaEnabled) {
+            if (!$authCode) {
```

Comment:
```
if (!$isMfaEnabled) {
    return;
}
```
And maybe the var is useless 
`if (!$user->isMfaEnabled || !$this->appli2faEnabled) `